src/services/refactors/moveToNewFile.ts

```diff
@@ -126,25 +139,39 @@ namespace ts.refactor {
 
         deleteUnusedOldImports(oldFile, toMove.all, changes, usage.unusedImportsFromOldFile, checker);
         deleteMovedStatements(oldFile, toMove.ranges, changes);
-
         updateImportsInOtherFiles(changes, program, oldFile, usage.movedSymbols, newModuleName);
 
         const imports = getNewFileImportsAndAddExportInOldFile(oldFile, usage.oldImportsNeededByNewFile, usage.newFileImportsFromOldFile, changes, checker, useEs6ModuleSyntax, quotePreference);
         const body = addExports(oldFile, toMove.all, usage.oldFileImportsFromNewFile, useEs6ModuleSyntax);
         if (imports.length && body.length) {
             return [
+                ...prologueDirectives,
                 ...imports,
                 SyntaxKind.NewLineTrivia as const,
                 ...body
             ];
         }
 
         return [
+            ...prologueDirectives,
             ...imports,
             ...body,
         ];
     }
 
+    function getPrologueDirectives(statements: readonly Statement[]): PrologueDirective[] {
+        const prologueDirectives: PrologueDirective[] = [];
+        for (const statement of statements) {
+            if (isPrologueDirective(statement)) {
+                prologueDirectives.push(statement);

```

Comment:
If you want, consider adding

```ts
function takeWhile<T, U>(array: readonly T[], predicate: (element: T) => element is U): U[];
function takeWhile<T>(array: readonly T[], predicate: (element: T) => boolean): T[];
```

Consider giving it similar semantics to `filter` where it doesn't allocate unless necessary.